buildSrc/src/main/groovy/com/github/piasy/okbuck/model/Target.groovy

```diff
@@ -0,0 +1,178 @@
+import java.util.jar.JarFile
+
+/**
+ * A target is roughly equivalent to what can be built with gradle via the various assemble tasks.
+ *
+ * For a project with no flavors and three build types - debug, release and development,
+ * the possible variants are debug, release and development.
+ * For a project with flavors flavor1 and flavor2 and three build types - debug, release and
+ * development, the possible variants are flavor1Debug, flavor1Release, flavor1Development,
+ * flavor2Debug, flavor2Release, flavor2Development.
+ *
+ * This class encapsulates all the data related to a variant to generate config files.
+ */
+@ToString(includes = ['project', 'name'])
+@EqualsAndHashCode(includes = ["project", "name"])
+abstract class Target {
+
+    static final Set<String> APT_CONFIGURATIONS = ["apt", "provided"] as Set
+    static final String PROCESSOR_ENTRY =
+            "META-INF/services/javax.annotation.processing.Processor"
+
+    final Project project
+    final String name
+    final String identifier
+    final String path
+    final Set<String> sources = [] as Set
+    final Set<Target> targetAptDeps = [] as Set
+    final Set<Target> targetCompileDeps = [] as Set
+
+    protected final Project rootProject
+    protected final Set<ExternalDependency> externalAptDeps = [] as Set
+    protected final Set<ExternalDependency> externalCompileDeps = [] as Set
+
+    /**
+     * Constructor.
+     *
+     * @param project The project.
+     * @param name The target name.
+     */
+    Target(Project project, String name) {
+        this.project = project
+        this.name = name
+
+        identifier = project.path.replaceFirst(':', '')
+        path = identifier.replaceAll(':', '/')
+
+        rootProject = project.gradle.rootProject
+
+        sources.addAll(getAvailable(sourceDirs()))
+
+        extractConfigurations(APT_CONFIGURATIONS, externalAptDeps, targetAptDeps)
+        OkBuckExtension okbuck = rootProject.okbuck
+        targetAptDeps.retainAll(targetAptDeps.findAll { Target target ->
+            target.getProp(okbuck.annotationProcessors, null) != null
+        })
+
+        extractConfigurations(compileConfigurations(), externalCompileDeps, targetCompileDeps)
+    }
+
+    /**
+     * List of source directories.
+     */
+    protected abstract Set<File> sourceDirs()
+
+    /**
+     * List of compile configurations.
+     */
+    protected abstract Set<String> compileConfigurations()
+
+    Set<String> getAnnotationProcessors() {
+        Set<String> annotationProcessors = [] as Set
+        aptDeps.each { String aptDep ->
+            JarFile jar = new JarFile(new File(aptDep))
+            List aptEntries = jar.entries().findAll { JarEntry entry ->
+                entry.name.equals(PROCESSOR_ENTRY)
+            }
+            if (!aptEntries.empty) {
+                aptEntries.each { JarEntry aptEntry ->
+                    annotationProcessors.add(IOUtils.toString(jar.getInputStream(aptEntry)).trim())
```

Comment:
There are libraries declared multiple annotation processors inside its `META-INF/services/javax.annotation.processing.Processor` file, such as `com.github.frankiesardo:auto-parcel-processor:0.3.1`, this should work well:

```groovy
String[] processors = IOUtils.toString(jar.getInputStream(aptEntry)).trim().split("\\n")
processors.each { String processor ->
       annotationProcessors.add(processor)
}
```